src/lib/components/composites/settings/project-settings/general/ProjectNameSettings.svelte

```diff
@@ -0,0 +1,121 @@
+<script lang="ts">
+    import SettingsSection from "$lib/components/composites/settings/SettingsSection.svelte";
+    import Input from "$lib/components/composites/input/Input.svelte";
+    import { Button } from "$lib/components/primitives/button";
+    import { backendService } from "$lib/grpc-api";
+    import { onMount } from "svelte";
+    import { Project } from "$lib/model/api/project";
+    import { Schema } from "$lib/schemas";
+    import { generateFieldMask } from "protobuf-fieldmask";
+    import type { ApiError } from "$lib/model/general";
+    import ErrorAlert from "$lib/components/composites/utils/ErrorAlert.svelte";
+    import { invalidate } from "$app/navigation";
+
+    interface Props {
+        projectId: string;
+        loadingProject: Promise<Project>;
+    }
+
+    const { projectId, loadingProject }: Props = $props();
+
+    let projectName: string = $state("");
+
+    let projectNameInput: Input;
+
+    let updateProjectError: ApiError | undefined = $state(undefined);
+
+    /**
+     * Loading the project name on mount.
+     */
+    onMount(async () => {
+        loadingProject
+            .then((project) => {
+                projectName = project.name;
+            })
+            .catch((error) => {
+                updateProjectError = {
+                    errorTitle: "Something went wrong while loading the project name on mount.",
+                    errorDetails: error,
+                };
+                console.error(updateProjectError);
+            });
+    });
+
+    /**
+     * Changes the name of the project, if a non-blank project name is provided.
+     *
+     * Therefore, send a request to the backend to update the project with the
+     * provided name.
+     */
+    async function handleSubmit(event: Event) {
+        event.preventDefault();
+
+        const projectNameValid = projectNameInput.validate();
+        if (!projectNameValid) return;
+
+        const projectData: Partial<Project> = {
+            id: projectId,
+            name: projectNameInput.getValue(),
+        };
+        const fieldMask = generateFieldMask(projectData);
+
+        backendService
+            .updateProject({
+                project: Project.create(projectData),
+                mask: {
+                    paths: fieldMask,
+                },
+            })
+            ?.response.then(() => {
+                invalidate("data:getProjectById");
+            })
+            .catch((error) => {
+                updateProjectError = {
+                    errorTitle: "Something went wrong while updating the project name.",
+                    errorDetails: error,
+                };
+                console.error(updateProjectError);
```

Comment:
Let's change this to something more descriptive.
```suggestion
                console.error(`Couldn't update project: ${error}`);
```